Tasks/AzureResourceManagerTemplateDeploymentV3/operations/DeploymentScopeBase.ts

```diff
@@ -51,9 +63,20 @@ export class DeploymentScopeBase {
                 this.armClient.deployments.createOrUpdate(this.taskParameters.deploymentName, this.deploymentParameters, (error, result, request, response) => {
                     if (error) {
                         if(this.taskParameters.deploymentScope === "Resource Group" && error.code == "ResourceGroupNotFound" && retryCount > 0){
-                            return this.waitAndPerformAzureDeployment(retryCount);
+                            return this.waitAndPerformAzureDeployment(retryCount, spnName);
                         }
                         utils.writeDeploymentErrors(this.taskParameters, error);
+
+                        if(error.statusCode == 403) {
```

Comment:
please move this to a separate method named printServicePrincipalRoleAssignmentError().

Reply:
Done